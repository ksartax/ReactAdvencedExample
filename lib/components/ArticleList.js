import React from 'react';

import Article from './Article';

class ArticleList extends React.PureComponent {
    render() {
        return (
            <div> 
                {Object.values(this.props.articles).map(article => (
                    <table>
                        <tr> 
                            <td> 
                                <Article 
                                    key={article.id}
                                    article={article}
                                />
                            </td>
                            <td> 
                                <button data-id={article.id} > Kasuj </button>
                            </td>
                        </tr>
                    </table>    
                ))}
            </div>
        )
    }
}

export default ArticleList;