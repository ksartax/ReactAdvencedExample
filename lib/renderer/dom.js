import React, {Component} from 'react';
import ReactDOM from 'react-dom';

import App from '../components/App';
import StateApi from '../state-api/lib/index';

const store = new StateApi(window.store);

ReactDOM.render(<App store={store}/>, document.getElementById('root'));