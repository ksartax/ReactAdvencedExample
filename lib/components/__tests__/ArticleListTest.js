import React from 'react';
import ArticleList from '../ArticleList';
import renderer from 'react-test-renderer';
import 'babel-polyfill'

describe('ArticleList test', () => {
    const testProps = {
        articles: {
            a: { id: 'a' },
            b: { id: 'b' }
        },
        articleActions: {
            lookupAuthor: jest.fn(() => ({})),
            remove: jest.fn(() => ({}))
        }
    }

    let tree;

    beforeAll(() => {
        tree = renderer.create(
            <ArticleList 
                {...testProps}
            />
        ).toJSON();
    })

    it("poprawne renderowanie", () => {
        expect(tree).toMatchSnapshot();
    }) 
})