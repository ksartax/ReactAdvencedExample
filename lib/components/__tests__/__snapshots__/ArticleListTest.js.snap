// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList test poprawne renderowanie 1`] = `
<div>
  <table>
    <tr>
      <td>
        <div
          style={
            Object {
              "borderBottomColor": "#aaa",
              "borderBottomStyle": "solid",
              "borderBottomWidth": 1,
              "marginBottom": 10,
              "paddingBottom": 10,
            }
          }
        >
          <div
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
             
             
          </div>
          <div>
             
            Invalid Date
             
          </div>
          <div
            style={
              Object {
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
          >
            <a>
               
            </a>
          </div>
          <div
            style={
              Object {
                "paddingLeft": 20,
              }
            }
          >
             
             
          </div>
        </div>
      </td>
      <td>
        <button
          data-id="a"
          onClick={[MockFunction]}
        >
           Kasuj 
        </button>
      </td>
    </tr>
  </table>
  <table>
    <tr>
      <td>
        <div
          style={
            Object {
              "borderBottomColor": "#aaa",
              "borderBottomStyle": "solid",
              "borderBottomWidth": 1,
              "marginBottom": 10,
              "paddingBottom": 10,
            }
          }
        >
          <div
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
             
             
          </div>
          <div>
             
            Invalid Date
             
          </div>
          <div
            style={
              Object {
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
          >
            <a>
               
            </a>
          </div>
          <div
            style={
              Object {
                "paddingLeft": 20,
              }
            }
          >
             
             
          </div>
        </div>
      </td>
      <td>
        <button
          data-id="b"
          onClick={[MockFunction]}
        >
           Kasuj 
        </button>
      </td>
    </tr>
  </table>
</div>
`;
